hs_collection_resource/tests/test_collection.py

```diff
@@ -527,3 +536,83 @@ def test_versioning(self):
         self.resCollection.resources.clear()
         self.assertEqual(self.resCollection.resources.count(), 0)
         self.assertEqual(new_collection.resources.count(), 3)
+
+    def test_update_collection_coverages(self):
+        # collection has no coverages metadata by default
+        self.assertEqual(self.resCollection.metadata.coverages.count(), 0)
+        # add 2 resources without coverage metadata to collection
+        self.resCollection.resources.add(self.resGen1)
+        self.resCollection.resources.add(self.resGeoFeature)
+        self.assertEqual(self.resCollection.resources.count(), 2)
+        # calculate overall coverages
+        _update_collection_coverages(self.resCollection)
+        # no collection coverage
+        self.assertEqual(self.resCollection.metadata.coverages.count(), 0)
+        # update resGen1 coverage
+        metadata_dict = [{'coverage': {'type': 'period', 'value': {'name': 'Name for period coverage', 'start': '1/1/2016', 'end': '12/31/2016'}}},]
+        update_science_metadata(pk=self.resGen1.short_id, metadata=metadata_dict)
+        self.assertEqual(self.resGen1.metadata.coverages.count(), 1)
+        # calculate overall coverages
+        _update_collection_coverages(self.resCollection)
+        # collection should have 1 coverage metadata: period
+        self.assertEqual(self.resCollection.metadata.coverages.count(), 1)
+        self.assertEqual(self.resCollection.metadata.coverages.all()[0].type.lower(), 'period')
```

Comment:
Instead of querying 3 times for the same element (coverage of type 'period'), use filter to get the element and then test the element attribute values.